### PR TITLE
#228 [WIP] Make ID store name configurable

### LINKED ISSUE #228: ID_STORE_NAME table renamed from Titan
In migrating from Titan, I found (by way of Cassandra table metrics reporting) that the `titan_ids` table has changed to `janusgraph_ids`. I think if this table were used it would be a breaking change.

However, it appears to me that this table isn't actually used much if at all? It's defined under the constant ID_STORE_NAME, which seems unused here:
https://github.com/JanusGraph/janusgraph/search?utf8=%E2%9C%93&q=ID_STORE_NAME&type=

Either there should be a migration path for this table from Titan to JanusGraph, or it should not exist at all. So far migrating to JanusGraph from Titan seems to work but things like this make me worry.

### PR DESCRIPTION
**Pull request by amcp**
Fixes #228

### CODE REVIEW

File: janusgraph-core/src/main/java/org/janusgraph/graphdb/configuration/GraphDatabaseConfiguration.java [hunk 1]
```diff
@@ -718,6 +718,14 @@ public boolean apply(@Nullable Integer integer) {
 //    public static final int IDS_BLOCK_SIZE_DEFAULT = 10000;
 
     /**
+     * The name of the ID store. Currently this defaults to janusgraph_ids. You can override the ID store to
+     * facilitate migration from JanusGraph's predecessor, Titan. Previously, this KCVStore was named titan_ids.
+     */
+    public static final ConfigOption<String> IDS_STORE_NAME = new ConfigOption<>(IDS_NS,"store-name",
```
Comment (jerryjch) [inline janusgraph-core/src/main/java/org/janusgraph/graphdb/configuration/GraphDatabaseConfiguration.java]: Nit. Space
`IDS_NS,"store-name"`

File: janusgraph-core/src/main/java/org/janusgraph/diskstorage/Backend.java [hunk 2]
```diff
@@ -85,8 +85,6 @@
     public static final String EDGESTORE_NAME = "edgestore";
     public static final String INDEXSTORE_NAME = "graphindex";
 
-    public static final String ID_STORE_NAME = "janusgraph_ids";
```
Comment (jerryjch) [inline janusgraph-core/src/main/java/org/janusgraph/diskstorage/Backend.java]: Can we add a comment telling why ID_STORE_NAME is missing here and moved to other place? e..g To make Make ID store name configurable

**Comment by jerryjch** [comment]
Let's say in the documentation that this IDS_STORE_NAME is meant to be used only for backward compatibility with Titan, and warn against using this property explicitly in normal operations.

File: janusgraph-core/src/main/java/org/janusgraph/diskstorage/Backend.java (hunk 2 above)
Comment (amcp) [inline janusgraph-core/src/main/java/org/janusgraph/diskstorage/Backend.java]: done

File: janusgraph-core/src/main/java/org/janusgraph/graphdb/configuration/GraphDatabaseConfiguration.java (hunk 1 above)
Comment (amcp) [inline janusgraph-core/src/main/java/org/janusgraph/graphdb/configuration/GraphDatabaseConfiguration.java]: done

**Comment by srosenthal** [comment]
It looks like this commit never made it to the master branch, only the branch for the 0.1.1. Am I right in assuming master is meant to be the latest work-in-progress for 0.2.0?

**Pull request merged**